buster/docparser.py

```diff
@@ -100,12 +100,22 @@ def get_all_subsections(soup: BeautifulSoup) -> tuple[list[str], list[str], list
     return documents_df
 
 
-def write_documents(filepath: str, documents_df: pd.DataFrame):
-    documents_df.to_csv(filepath, index=False)
+def write_documents(filepath: str, documents_df: pd.DataFrame, format: str = "pickle"):
+    if format == "csv":
+        documents_df.to_csv(filepath, index=False)
+    elif format == "pickle":
+        documents_df.to_pickle(filepath)
+    else:
+        raise ValueError(f"Unsupported format: {format}. Accepted formats are: csv, pickle.")
 
 
-def read_documents(filepath: str) -> pd.DataFrame:
-    return pd.read_csv(filepath)
+def read_documents(filepath: str, format: str = "pickle") -> pd.DataFrame:
+    if format == "csv":
```

Comment:
can't we just infer the format from the file extension provided in the `filepath`?

Reply:
pickle supports a lot of compression algorithms ( ‘.gz’, ‘.bz2’, ‘.zip’, ‘.xz’, ‘.zst’, ‘.tar’, ‘.tar.gz’, ‘.tar.xz’ or ‘.tar.bz2’ ) and will infer based on the file extension. I didn't want to force any in particular.